Sources/SymbolKit/UnifiedSymbolGraph/UnifiedSymbolGraph.swift

```diff
@@ -160,6 +160,190 @@ extension UnifiedSymbolGraph {
+        //    with any relationships that referenced them.
+
+        /// A mapping of overload group symbols to their individual overloads.
+        var overloadGroups: [String: Set<String>] = [:]
+        /// A mapping of overloaded symbols to the overload groups they are part of.
+        var overloadGroupsBySymbol: [String: Set<String>] = [:]
+
+        for relationship in self.relationshipsByLanguage.flatMap(\.value)
+            where relationship.kind == SymbolGraph.Relationship.Kind.overloadOf
+        {
+            overloadGroups[relationship.target, default: []].insert(relationship.source)
+            overloadGroupsBySymbol[relationship.source, default: []].insert(relationship.target)
+        }
+
+        for overloadGroup in overloadGroups.keys {
+            guard overloadGroups.keys.contains(overloadGroup) else {
+                // If this overload group was a sibling to another overload group, we'll have
+                // removed it from the dictionary, so don't try to process it again.
+                continue
+            }
+            var siblingOverloadGroups: Set<String> = []
+            var overloadedSymbols: Set<String> = []
+            var pendingOverloadGroups: Set<String> = [overloadGroup]
+
+            // 1. Collect all the overloaded symbols and all the overload groups they belong to.
+            while !pendingOverloadGroups.isEmpty {
+                for processingOverloadGroup in pendingOverloadGroups {
+                    pendingOverloadGroups.remove(processingOverloadGroup)
+                    siblingOverloadGroups.insert(processingOverloadGroup)
+
+                    for overloadedSymbol in overloadGroups[processingOverloadGroup]! {
+                        overloadedSymbols.insert(overloadedSymbol)
+                        for otherOverloadGroup in overloadGroupsBySymbol[overloadedSymbol]! {
+                            if !siblingOverloadGroups.contains(otherOverloadGroup),
+                               !pendingOverloadGroups.contains(otherOverloadGroup)
+                            {
+                                pendingOverloadGroups.insert(otherOverloadGroup)
+                            }
+                        }
+                    }
+                }
+            }
+
+            guard !overloadedSymbols.isEmpty else { continue }
+
+            // 2. Sort the overloaded symbols with the same comparator as `SymbolGraph.createOverloadGroupSymbols()`.
+            var allSymbolsHaveDeclaration = true
+            let sortedOverloads = overloadedSymbols.map({ identifier in
+                let symbol = symbols[identifier]!
+                if !symbol.mixins.values.contains(where: { $0.keys.contains(SymbolGraph.Symbol.DeclarationFragments.mixinKey) }) {
+                    allSymbolsHaveDeclaration = false
+                }
+                return symbol
+            }).sorted(by: Symbol.sortForOverloads(orderByDeclaration: allSymbolsHaveDeclaration))
```

Comment:
This is a bit confusing to read  - maybe set `allSymbolsHaveDeclaration` directly using `allSatisfy` and break things up a bit... something like:

```suggestion
            let overloads = overloadedSymbols.map{ symbols[$0]! }
            let allSymbolsHaveDeclaration = overloads.allSatisfy {
                $0.mixins.values.contains(
                    where: { $0.keys.contains(SymbolGraph.Symbol.DeclarationFragments.mixinKey) }
                )
            }
            let sortedOverloads = overloads.sorted(by: Symbol.sortForOverloads(orderByDeclaration: allSymbolsHaveDeclaration))
```

... although writing this makes me think we might want to rename `overloadedSymbols` to be `overloadedSymbolIdentifiers`. It's a bit confusing to have to distinguish between symbols, overloadedSymbols etc. when one is a symbol and the other a string identifier.

Reply:
This way i only had to iterate over the overloads once, but i don't see a single overload group having a ton of symbols, so i can clean this up like you suggest. 👍